protocol/rpcconsumer/rpcconsumer.go

```diff
@@ -247,10 +248,25 @@ rpcconsumer 127.0.0.1:3333 COS3 tendermintrpc 127.0.0.1:3334 COS3 rest <flags>`,
 			}
 			// handle flags, pass necessary fields
 			ctx := context.Background()
-			networkChainId, err := cmd.Flags().GetString(flags.FlagChainID)
+			var networkChainId string
+			cmdChainIdFlag := viper.GetString(flags.FlagChainID)
+			clientTomlConfig, err := config.ReadFromClientConfig(clientCtx)
 			if err != nil {
 				return err
```

Comment:
lets not kill the process if this method failed. just print an error and continue without it

Reply:
you can basically add the functionality inside 

if err == nil {
    functionality 
}

also we want to read config only if chainId was not specified so 

if cmdChainIdFlag == app.Name {
   clientTomlConfig, err := config.ReadFromClientConfig(clientCtx)
   if err == nil {
       functionality 
   }
}